File: docs/archetype_document_templates.py
from enum import Enum
from pydantic import BaseModel
from typing import List

class DocumentVisibility(str, Enum):
    PUBLIC = 'public'
    STEWARD_ONLY = 'steward_only'
    LINEAGE_RESTRICTED = 'lineage_restricted'

class ReplayBindingMode(str, Enum):
    APPEND_ONLY = 'append_only'
    LINEAGE_LINKED = 'lineage_linked'

class RegistryVersion(BaseModel):
    registry_version: str
    last_updated: str
    governance_revision: str

ARCHETYPE_DOCUMENT_TEMPLATES_VERSION = RegistryVersion(
    registry_version="1.0.0",
    last_updated="2026-05-27",
    governance_revision="REV-001"
)

class ArchetypeDocumentTemplate(BaseModel):
    template_key: str
    archetype_key: str
    document_type: str
    title_pattern: str
    sections: List[str]
    visibility_default: DocumentVisibility
    replay_binding_mode: ReplayBindingMode
    governance_boundary: str

ARCHETYPE_DOCUMENT_TEMPLATES = [
    ArchetypeDocumentTemplate(
        template_key='TPL-MIN-001',
        archetype_key='community_ministry_steward',
        document_type='ministry_teaching',
        title_pattern='Teaching Scroll - {Title}',
        sections=[
            'SEC-ID-001',
            'SEC-REFL-001',
            'SEC-CONT-001'
        ],
        visibility_default=DocumentVisibility.PUBLIC,
        replay_binding_mode=ReplayBindingMode.APPEND_ONLY,
        governance_boundary='Teaching preserves declared interpretation. It does not manufacture spiritual authority.'
    ),
    ArchetypeDocumentTemplate(
        template_key='TPL-FIN-001',
        archetype_key='financial_insurance_steward',
        document_type='business_proposal',
        title_pattern='Family Cover Proposal - {ClientName}',
        sections=[
            'SEC-ID-001',
            'SEC-PRICE-001',
            'SEC-SIG-001'
        ],
        visibility_default=DocumentVisibility.LINEAGE_RESTRICTED,
        replay_binding_mode=ReplayBindingMode.LINEAGE_LINKED,
        governance_boundary='Proposal reflects declared offering. It does not guarantee outcome or final underwriter approval.'
    ),
    ArchetypeDocumentTemplate(
        template_key='TPL-RET-001',
        archetype_key='local_retail_steward',
        document_type='catalogue',
        title_pattern='Specials Catalogue - {Month}',
        sections=[
            'SEC-ID-001',
            'SEC-PRICE-001'
        ],
        visibility_default=DocumentVisibility.PUBLIC,
        replay_binding_mode=ReplayBindingMode.APPEND_ONLY,
        governance_boundary='Catalogue declares current availability. It does not guarantee future stock or indefinite pricing.'
    )
]

File: archive_apps/mobile/data/archetypeDocumentTemplates.ts
// Governed registry of archetype-specific document templates for iPhande
// Enforces structured, replay-aware, and governance-aware long-form continuity generation.

export type DocumentVisibility = 'public' | 'steward_only' | 'lineage_restricted';
export type ReplayBindingMode = 'append_only' | 'lineage_linked';

export interface ArchetypeDocumentTemplate {
    templateKey: string;
    archetypeKey: string;
    documentType: string;
    titlePattern: string;
    sections: string[];
    visibilityDefault: DocumentVisibility;
    replayBindingMode: ReplayBindingMode;
    governanceBoundary: string;
}

export const ARCHETYPE_DOCUMENT_TEMPLATES: ArchetypeDocumentTemplate[] = [
    // --- MINISTRY & COMMUNITY STEWARD ---
    {
        templateKey: 'TPL-MIN-001',
        archetypeKey: 'community_ministry_steward',
        documentType: 'ministry_teaching',
        titlePattern: 'Teaching Scroll - {Title}',
        sections: [
            'Title',
            'Scripture Foundation',
            'Teaching Body',
            'Reflection',
            'Prayer',
            'Continuity Notes'
        ],
        visibilityDefault: 'public',
        replayBindingMode: 'append_only',
        governanceBoundary: 'Teaching preserves declared interpretation. It does not manufacture spiritual authority.'
    },

    // --- FINANCIAL & INSURANCE STEWARD (Funeral Cover) ---
    {
        templateKey: 'TPL-FIN-001',
        archetypeKey: 'financial_insurance_steward',
        documentType: 'business_proposal',
        titlePattern: 'Family Cover Proposal - {ClientName}',
        sections: [
            'Client Details',
            'Proposed Cover',
            'Terms Overview',
            'Steward Commitment',
            'Next Steps'
        ],
        visibilityDefault: 'lineage_restricted',
        replayBindingMode: 'lineage_linked',
        governanceBoundary: 'Proposal reflects declared offering. It does not guarantee outcome or final underwriter approval.'
    },

    // --- LOCAL RETAIL STEWARD ---
    {
        templateKey: 'TPL-RET-001',
        archetypeKey: 'local_retail_steward',
        documentType: 'catalogue',
        titlePattern: 'Specials Catalogue - {Month}',
        sections: [
            'Store Context',
            'Current Specials',
            'Standard Items',
            'Operating Hours',
            'Contact Information'
        ],
        visibilityDefault: 'public',
        replayBindingMode: 'append_only',
        governanceBoundary: 'Catalogue declares current availability. It does not guarantee future stock or indefinite pricing.'
    },

    // --- EDUCATION & TUTORING STEWARD ---
    {
        templateKey: 'TPL-EDU-001',
        archetypeKey: 'education_tutoring_steward',
        documentType: 'study_guide',
        titlePattern: 'Learning Guide - {Topic}',
        sections: [
            'Topic Overview',
            'Core Concepts',
            'Practice Exercises',
            'Review Questions',
            'Tutor Notes'
        ],
        visibilityDefault: 'public',
        replayBindingMode: 'append_only',
        governanceBoundary: 'Guide provides structured knowledge. It does not guarantee academic results or official accreditation.'
    },

    // --- CREATIVE & MEDIA STEWARD ---
    {
        templateKey: 'TPL-CRE-001',
        archetypeKey: 'creative_media_steward',
        documentType: 'portfolio_scroll',
        titlePattern: 'Portfolio Scroll - {ProjectName}',
        sections: [
            'Project Context',
            'Media Showcase',
            'Steward Role',
            'Client Outcome',
            'Contact for Booking'
        ],
        visibilityDefault: 'public',
        replayBindingMode: 'append_only',
        governanceBoundary: 'Portfolio presents past continuity. It does not predict exact future deliverables.'
    }
];
